### PR TITLE
(#2642) Remove deprecated shims

### PR DESCRIPTION
**Pull request by gep13**
## Description Of Changes

The shims that were previously marked as deprecated have now been removed.

Also the documentation around the deprecations have been removed, and tests updated to reflect that these shims are no longer included.

## Motivation and Context

The shims for chocolatey, cinst, clist, cpush, cuninst, and cup, where deprecated in a previous release of Chocolatey, and this commit removes them, so that they will not appear in the upcoming 2.0.0 release.

## Testing

1. Ran build to ensure all tests pass, both unit and integration
2. Installed the generated package from the build and verified that everything worked as expected, and shims weren't included

### Operating Systems Testing

- Windows 10/11

## Change Types Made

* [] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [x] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added. (Tests has not been updated, as I do not know which are affected by localization and which may have been affected by the update, tests will instead be enabled or added to Chocolatey CLI E2E)
* [x] All new and existing tests passed? (More tests has been made English only since the last update, unable to verify whether they succeed or not)
* [ ] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issues

- #2642  
- https://app.clickup.com/t/20540031/PROJ-527

### CODE REVIEW

File: src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs
Comment (vexx32) [inline src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs:129]: This doesn't feel like it should be related to this PR? 🤔 In combination with other deprecations, absolutely, but on its own shouldn't this PR leave in the one deprecation message specifically about `choco list --local-only`?

**Review by vexx32 — APPROVED** [review]
One minor comment, not sure it really matters, but otherwise this looks good to me. Thanks for sorting this out! 💖

File: src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs
Comment (gep13) [reply src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs:129]: I had the same thought, but that felt like the wrong thing to do, given that we know that this work is coming out, and I was in that code anyway.  Looks like Kim is removing the same thing in his PR as well, so I am happy to rebase this work once his PR is merged.

File: setup.ps1
Comment (corbob) [inline setup.ps1:1]: What even is this file? I get that it installs Chocolatey, but then it installs a version of pester that is downright ancient...

File: tests/chocolatey-tests/chocolatey.Tests.ps1
Comment (corbob) [inline tests/chocolatey-tests/chocolatey.Tests.ps1:57]:
```suggestion
        It "has <_> available on PATH" -ForEach @("choco.exe") {
```
I can't believe @vexx32 didn't catch this 😂

File: tests/chocolatey-tests/commands/choco-deprecated.Tests.ps1
Comment (corbob) [inline tests/chocolatey-tests/commands/choco-deprecated.Tests.ps1:1]: Should this file be removed between this and @vexx32's PR to remove `Get-BinRoot`?

**Review by corbob — CHANGES_REQUESTED** [review]
One thing that will need to change, and two questions that do not block merging at all.

File: tests/chocolatey-tests/commands/choco-deprecated.Tests.ps1
Comment (vexx32) [reply tests/chocolatey-tests/commands/choco-deprecated.Tests.ps1:1]: Maybe? dunno

Comment (corbob) [reply tests/chocolatey-tests/commands/choco-deprecated.Tests.ps1:1]: yup, between the two of them, the only thing left would have been the module import at the top 😂 I've rebased this and resolved the merge conflict.

File: tests/chocolatey-tests/chocolatey.Tests.ps1
Comment (corbob) [reply tests/chocolatey-tests/chocolatey.Tests.ps1]: Edited/rebased this into the existing commit and force pushed.

**Review by corbob — APPROVED** [review]

File: src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs
Comment (corbob) [reply src/chocolatey.tests/infrastructure.app/commands/ChocolateyListCommandSpecs.cs:129]: Going to merge this one, and rebase into Kim's since that's the order we're at right now...

**Review by corbob — APPROVED** [review]

File: setup.ps1
Comment (corbob) [reply setup.ps1:1]: Apparently can't merge without resolving conversations. So I'll drop this on Slack because it doesn't need us to deal with it now.

**Comment by corbob** [comment]
Thanks for getting this sorted @gep13 👍

**Pull request merged**
